math/sum_of_digits.cpp

```diff
@@ -0,0 +1,28 @@
+/**
+ * A simple C++ Program to find the Sum of Digits of input integer.
+ */
+#include <iostream>
+
+/**
+ * Function to find the sum of the digits of an integer.
+ * @param num The integer.
+ * @return Sum of the digits of the integer.
+ */
+int sum_of_digits(int num) {
+    int sum = 0;
+    while (num > 0) {
+        sum = sum + (num % 10);
+        num = num / 10;
+    }
+    return sum;
+}
+
+int main() {
+    int num;
+    std::cin >> num;
+    if (num < 0) {
+        num = -1 * num;
+    }
+    std::cout << sum_of_digits(num);
```

Comment:
please add a test case and self-validation:

- create a function void test()

- use a pre-defined number

> say 199765


- call sum_of_digits(int num) function to illustrate its call

- use assert statements to verify the results are as expected.

This ensures the the program is self-reliant and to high-standards due to self-validation. More tests the better